src/Aspire.Hosting.Testing/DistributedApplicationFactory.cs

```diff
@@ -131,16 +134,66 @@ private void OnBuilderCreatingCore(DistributedApplicationOptions applicationOpti
         applicationOptions.AssemblyName = _entryPoint.Assembly.GetName().Name ?? string.Empty;
         applicationOptions.DisableDashboard = true;
         var cfg = hostBuilderOptions.Configuration ??= new();
-        cfg.AddInMemoryCollection(new Dictionary<string, string?>
+        var additionalConfig = new Dictionary<string, string?>
         {
             ["DcpPublisher:RandomizePorts"] = "true",
             ["DcpPublisher:DeleteResourcesOnShutdown"] = "true",
             ["DcpPublisher:ResourceNameSuffix"] = $"{Random.Shared.Next():x}",
-        });
+        };
+
+        var appHostProjectPath = ResolveProjectPath(_entryPoint.Assembly);
+        if (!string.IsNullOrEmpty(appHostProjectPath))
+        {
+            hostBuilderOptions.ContentRootPath = appHostProjectPath;
+        }
+
+        var appHostLaunchSettings = GetLaunchSettings(appHostProjectPath);
+        if (appHostLaunchSettings?.Profiles.FirstOrDefault().Key is { } profileName)
+        {
+            additionalConfig["AppHost:DefaultLaunchProfileName"] = profileName;
```

Comment:
Wonder whether we should start putting these keys in constants somewhere.

Reply:
We probably should do a sweep of them at some point